subsys/partition_manager/CMakeLists.txt

```diff
@@ -45,3 +45,21 @@ if((EXISTS ${APPLICATION_SOURCE_DIR}/pm.yml) AND IMAGE)
     ${IMAGE}
     )
 endif()
+
+get_property(img_pm_subsys GLOBAL PROPERTY ${IMAGE}PM_SUBSYS)
+if (img_pm_subsys)
+  foreach (subsys_pm_yml ${img_pm_subsys})
+    file(RELATIVE_PATH rel_path_to_yml ${ZEPHYR_BASE} ${subsys_pm_yml})
+    set(subsys_pm_preprocessed
+      ${PROJECT_BINARY_DIR}/${rel_path_to_yml})
```

Comment:
I think this causes duplicate pm.yml files (if multiple images include the same subsys) to be placed in the same location, overwriting one another. Maybe add an ${IMAGE} path element to make sure there's no overwriting happening?

Or I might be wrong about the whole thing :P.

Reply:
This is a documented limitation:

```
There are some limitations when multiple images include the same subsystem which defines a Partition Manager configuration.
All partitions are global, and will only be included once, even if multiple configurations define them.
When multiple configurations define the same partition, the configuration of that partition must be identical in all definitions.
An exception is raised if the configuration differs.
```

Reply:
But if I'm right, "An exception is raised if the configuration differs." is not correct, since there will be only one configuration file, which has been overwritten a number of times, so whichever was last "wins"

Reply:
AFAICT the preprocessed pm.yml files are stored in different build output directories (per image) and the partition_manager script contains code verifying that a duplicate definition is identical.

Reply:
Sorry, I think you're right. This code is run once for each image, so PROJECT_BINARY_DIR will have different values for each. I thought it was run just once globally.